src/combinatorics/quiver/single_obj.lean

```diff
@@ -0,0 +1,121 @@
+/-
+Copyright (c) 2022 Antoine Labelle. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Antoine Labelle
+-/
+import combinatorics.quiver.cast
+import combinatorics.quiver.symmetric
+
+/-!
+# Single-object quiver
+
+Single object quiver with a given arrows type.
+
+## Main definitions
+
+Given a type `α`, `single_obj α` is the `unit` type, whose single object is called `star α`, with
+`quiver` structure such that `star α ⟶ star α` is the type `α`.
+An element `x : α` can be reinterpreted as an element of `star α ⟶ star α` using
+`to_hom`.
+More generally, a list of elements of `a` can be reinterpreted as a path from `star α` to
+itself using `path_equiv_list`.
+-/
+
+namespace quiver
+
+/-- Type tag on `unit` used to define single-object quivers. -/
+@[nolint unused_arguments]
+def single_obj (α : Type*) : Type := unit
+
+namespace single_obj
+
+variables (α β γ : Type*)
+
+instance : quiver (single_obj α) := ⟨λ _ _, α⟩
+
+/-- The single object in `single_obj α`. -/
+def star : single_obj α := unit.star
+
+instance : inhabited (single_obj α) := ⟨star α⟩
+
+variables {α β γ}
+
+/-- Equip `single_obj α` with a reverse operation. -/
+@[reducible] -- See note [reducible non-instances]
+def has_reverse (rev : α → α) : has_reverse (single_obj α) := ⟨λ _ _, rev⟩
+
+/-- Equip `single_obj α` with an involutive reverse operation. -/
+@[reducible] -- See note [reducible non-instances]
+def has_involutive_reverse (rev : α → α) (h : function.involutive rev) :
+  has_involutive_reverse (single_obj α) :=
+{ to_has_reverse := has_reverse rev,
+  inv' := λ _ _, h}
+
+/-- The type of arrows from `star α` to itself is equivalent to the original type `α`. -/
+@[simps] def to_hom : α ≃ (star α ⟶ star α) := equiv.refl _
+
+/--
+Prefunctors between two `single_obj` quivers correspond to functions between the corresponding
+arrows types.
+-/
+def to_prefunctor :
```

Comment:
Maybe you could replace `to_prefunctor_symm_id` and `to_prefunctor_symm_comp` by a single lemma `to_prefunctor_symm_apply`?

Reply:
I added `@[simps]` to `to_prefunctor` which generates `to_prefunctor_symm_apply`, but I still believe that `to_prefunctor_symm_comp` is relevant since we might want eventually to make `single_obj` a functor. I removed the simp attribute though, does that seem good?

Reply:
I think you can just get rid of `to_prefunctor_symm_comp`. It is easily provable by simp now.

Also can you use `@[simps?]` to check it generates lemmas in terms of `coe_fn`, not `to_fun`?

Reply:
I tried to check if `to_prefunctor_symm_comp` is closed by `by {ext, simp}` and apparently it's not. simp leaves the goal  `g.map (f.map x) = g.map (f.map x)`, which somehow isn't proved by `rfl`. Do you have an idea what's the issue?

Reply:
`@[simps]` generates the following lemma :
`⇑(to_prefunctor.symm) f a = f.map (⇑to_hom a)`

Reply:
Actually, `to_prefunctor_symm_apply` is applied one too many time to prove `to_prefunctor_symm_comp`, so just tag `to_prefunctor_symm_comp` simp again.